src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c

```diff
@@ -101,122 +158,1003 @@ static DamageTable D_80B7346C = {
+                       Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_22C->actor.world.pos), 1, 0x71C, 0xB6);
+
+    if ((this->actor.bgCheckFlags & 1) && (this->unk_22C->actor.colChkInfo.health == 0)) {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    } else {
+        this->actor.velocity.y += this->actor.gravity;
+        this->actor.world.pos.y += this->actor.velocity.y;
+        Math_SmoothStepToF(&this->actor.world.pos.x, this->unk_22C->actor.world.pos.x, 0.6f, 1.6f, 0.1f);
+        Math_SmoothStepToF(&this->actor.world.pos.z, this->unk_22C->actor.world.pos.z, 0.6f, 1.6f, 0.1f);
+    }
+
+    if (this->actor.bgCheckFlags & 2) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if ((sp34 < 50.0f) && (this->actor.bgCheckFlags & 1)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALKID_ATTACK);
+        func_800BDC5C(&this->skelAnime, sAnimations, 9);
+        this->actor.velocity.y = 10.0f;
+
+        for (i = 0; i < 4; i++) {
+            func_80B72430(this, globalCtx, 0);
+        }
+    }
+}
+
+void func_80B71D8C(EnRailSkb* this, GlobalContext* globalCtx, EnRailSkbUnkFunc unkFunc) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_ENEMY].first;
+
+    while (actor != NULL) {
+        if (actor->id == ACTOR_EN_RAIL_SKB) {
+            unkFunc((EnRailSkb*)actor);
+        }
+        actor = actor->next;
+    }
+}
+
+void func_80B71DF0(EnRailSkb* this) {
+    if (this->actionFunc == func_80B714D8) {
+        this->actor.shape.yOffset = -5000.0f;
+        this->actor.colChkInfo.health = 2;
+        this->unk_402 = 0;
+        this->actor.flags |= 1;
+        this->actor.draw = EnRailSkb_Draw;
+        this->actor.shape.shadowAlpha = 0;
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+    } else if ((this->actionFunc == func_80B7114C) && (this->unk_3F0 >= 2)) {
+        this->unk_3F0 = 0;
+    }
+
+    func_800BDC5C(&this->skelAnime, sAnimations, 11);
+    this->actionFunc = func_80B71EA8;
+}
+
+void func_80B71EA8(EnRailSkb* this, GlobalContext* globalCtx) {
+    if (Math_SmoothStepToF(&this->actor.shape.yOffset, 0.0f, 0.5f, 500.0f, 10.0f) != 0.0f) {
+        func_80B72430(this, globalCtx, 1);
+    }
+
+    if (this->actor.shape.shadowAlpha != 255) {
+        if (this->actor.shape.shadowAlpha >= 235) {
+            this->actor.shape.shadowAlpha = 255;
+        } else {
+            this->actor.shape.shadowAlpha += 20;
+        }
+    }
+}
+
+void func_80B71F3C(EnRailSkb* this, GlobalContext* globalCtx) {
+    if (((this->unk_400 == 5101) || (this->unk_400 == 5109)) && (this->unk_3F2 != 1)) {
+        func_80B71D8C(this, globalCtx, func_80B71DF0);
+        func_80B717C8(this);
+        this->unk_3F2 = 1;
+    }
+
+    if (func_80147624(globalCtx)) {
+        switch (this->unk_400) {
+            case 5100:
+                func_801518B0(globalCtx, 5101, &this->actor);
+                this->unk_400 = 5101;
+                break;
```

Comment:
Those are `textId`s. We use hex for them.
```suggestion
        switch (this->unk_400) {
            case 0x13EC:
                func_801518B0(globalCtx, 0x13ED, &this->actor);
                this->unk_400 = 0x13ED;
                break;
```
Please change every use of `unk_400` accordingly